codebuild_specs/scripts-windows/shared-scripts-windows.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash
+
+# set exit on error to true
+set -e
+
+# The flags address the issue here: https://github.com/boto/botocore/issues/1716
+export MSYS_NO_PATHCONV=1
+export MSYS2_ARG_CONV_EXCL="*"
+
+# We have custom caching for our CodeBuild pipelines
+# which allows us to share caches with jobs in the same batch
+
+# storeCache <local path> <cache location>
+function storeCache {
+    localPath="$1"
+    alias="$2"
+    s3Path="s3://$CACHE_BUCKET_NAME/$CODEBUILD_SOURCE_VERSION/$alias"
+    echo writing cache to $s3Path
+    # zip contents and upload to s3
+    # if ! (cd $localPath && tar -czf cache.tar . && ls && aws s3 cp cache.tar $s3Path); then
+    #     echo Something went wrong storing the cache.
+    # fi
+    if ! (cd $localPath && tar -czf cache.tar . && ls && aws s3 cp cache.tar $s3Path); then
+        echo Something went wrong storing the cache.
+    fi
+    echo done writing cache
```

Comment:
Will this work?

Reply:
The echo will work, the syntax highlighter is just being a little overzealous.

Reply:
yeah it works